<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>702</width>
    <height>479</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="pBnone">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>-20</y>
      <width>16</width>
      <height>441</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QFrame" name="inputdata">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>0</y>
      <width>521</width>
      <height>81</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>501</width>
       <height>61</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout2">
      <item row="0" column="0">
       <widget class="QLabel" name="labelDatapath">
        <property name="text">
         <string>data path:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QComboBox" name="cBDatadir"/>
      </item>
      <item row="1" column="1">
       <widget class="QToolButton" name="tBChoosedir">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gBToolPara">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>131</width>
      <height>131</height>
     </rect>
    </property>
    <property name="baseSize">
     <size>
      <width>3</width>
      <height>0</height>
     </size>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox {
border: 1px solid gray;
border-radius: 9px;
margin-top: 0.5em;
}

QGroupBox::title {
subcontrol-origin: margin;
left: 10px;
padding: 0 3px 0 3px;
}</string>
    </property>
    <property name="title">
     <string>Tool (mm)</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>31</width>
       <height>101</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout4">
      <item row="1" column="1">
       <widget class="QLabel" name="labelTy">
        <property name="text">
         <string>Ty</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="labelTx">
        <property name="text">
         <string>Tx</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="labelTz">
        <property name="text">
         <string>Tz</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>61</width>
       <height>101</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout3">
      <item row="2" column="0">
       <widget class="QLineEdit" name="lETy">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0</string>
        </property>
        <property name="echoMode">
         <enum>QLineEdit::Normal</enum>
        </property>
        <property name="cursorPosition">
         <number>0</number>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLineEdit" name="lETz">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0</string>
        </property>
        <property name="cursorPosition">
         <number>0</number>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="lETx">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>0</string>
        </property>
        <property name="cursorPosition">
         <number>0</number>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>0</y>
      <width>111</width>
      <height>61</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="RobotType">
     <item row="2" column="0">
      <widget class="QComboBox" name="cBRobotType">
       <item>
        <property name="text">
         <string>I3</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>I5</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>I7</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>I10</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labelRobotType">
       <property name="text">
        <string>RobotType</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="gBMeasurePara">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>270</y>
      <width>131</width>
      <height>131</height>
     </rect>
    </property>
    <property name="baseSize">
     <size>
      <width>3</width>
      <height>0</height>
     </size>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox {
border: 1px solid gray;
border-radius: 9px;
margin-top: 0.5em;
}

QGroupBox::title {
subcontrol-origin: margin;
left: 10px;
padding: 0 3px 0 3px;
}</string>
    </property>
    <property name="title">
     <string>Measure (mm)</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>31</width>
       <height>101</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayoutMP">
      <item row="1" column="1">
       <widget class="QLabel" name="labelMy">
        <property name="text">
         <string>My</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="labelMx">
        <property name="text">
         <string>Mx</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="labelMz">
        <property name="text">
         <string>Mz</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget_6">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>61</width>
       <height>101</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayoutMPD">
      <item row="2" column="0">
       <widget class="QLineEdit" name="lEMy">
        <property name="text">
         <string>0</string>
        </property>
        <property name="cursorPosition">
         <number>0</number>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLineEdit" name="lEMz">
        <property name="text">
         <string>0</string>
        </property>
        <property name="cursorPosition">
         <number>0</number>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="lEMx">
        <property name="text">
         <string>0</string>
        </property>
        <property name="cursorPosition">
         <number>0</number>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gBCalibrationPara">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>100</y>
      <width>471</width>
      <height>241</height>
     </rect>
    </property>
    <property name="baseSize">
     <size>
      <width>3</width>
      <height>0</height>
     </size>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox {
border: 1px solid gray;
border-radius: 9px;
margin-top: 0.5em;
}

QGroupBox::title {
subcontrol-origin: margin;
left: 10px;
padding: 0 3px 0 3px;
}</string>
    </property>
    <property name="title">
     <string>Calibration Para (degree/mm)</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>360</width>
       <height>194</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gLDHpara">
      <item row="1" column="0">
       <widget class="QPushButton" name="pBalpha2">
        <property name="text">
         <string>alpha2</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="pBtheta1">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>theta1</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="pBtheta2">
        <property name="text">
         <string>theta2</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QPushButton" name="pBalpha5">
        <property name="text">
         <string>alpha5</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="pBa3">
        <property name="text">
         <string>a3</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QPushButton" name="pBtheta6">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>theta6</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="pBtheta3">
        <property name="text">
         <string>theta3</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="pBalpha1">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>alpha1</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QPushButton" name="pBtheta4">
        <property name="text">
         <string>theta4</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="pBa1">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>a1</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QPushButton" name="pBalpha6">
        <property name="text">
         <string>alpha6</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="pBa2">
        <property name="text">
         <string>a2</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QPushButton" name="pBa6">
        <property name="text">
         <string>a6</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="pBa4">
        <property name="text">
         <string>a4</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QPushButton" name="pBa5">
        <property name="text">
         <string>a5</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="pBalpha4">
        <property name="text">
         <string>alpha4</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="pBalpha3">
        <property name="text">
         <string>alpha3</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="pBd1">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>d1</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QPushButton" name="pBd6">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>d6</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QPushButton" name="pBd5">
        <property name="text">
         <string>d5</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QPushButton" name="pBd4">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>d4</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QPushButton" name="pBd3">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>d3</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QPushButton" name="pBd2">
        <property name="text">
         <string>d2</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QPushButton" name="pBtheta5">
        <property name="text">
         <string>theta5</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget_8">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>30</y>
       <width>91</width>
       <height>194</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gLBetapara">
      <item row="2" column="0">
       <widget class="QPushButton" name="pBbeta3">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>beta3</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QPushButton" name="pBbeta5">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>beta5</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="pBbeta1">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>beta1</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QPushButton" name="pBbeta6">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>beta6</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="pBbeta4">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>beta4</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="pBbeta2">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>beta2</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget_9">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>240</y>
       <width>91</width>
       <height>71</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout"/>
    </widget>
   </widget>
   <widget class="QCheckBox" name="cBCalibrationBeta">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>CalibrationBeta</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>360</y>
      <width>201</width>
      <height>40</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="pBrestart">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>restart</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pBoutputresult">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>output result</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>360</y>
      <width>121</width>
      <height>52</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QRadioButton" name="rBshowDpara">
       <property name="text">
        <string>show dpara</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="rBshowpara">
       <property name="text">
        <string>show para</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pBCalibrate">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>350</y>
      <width>101</width>
      <height>60</height>
     </rect>
    </property>
    <property name="text">
     <string>Start</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>702</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
